crypto.c

```diff
@@ -517,20 +608,39 @@ dtls_ecdsa_verify_sig_hash(const unsigned char *pub_key_x,
 			   const unsigned char *pub_key_y, size_t key_size,
 			   const unsigned char *sign_hash, size_t sign_hash_size,
 			   unsigned char *result_r, unsigned char *result_s) {
+  const dtls_ecdh_curve curve = default_curve;
   uint8_t pub_key[2 * DTLS_EC_KEY_SIZE];
+  memcpy(pub_key, pub_key_x, key_size);
+  memcpy(pub_key + key_size, pub_key_y, key_size);
+  return dtls_ecdsa_verify_sig_hash2(pub_key, key_size,
+                                     sign_hash, sign_hash_size,
+                                     curve,
+                                     result_r, result_s);
+}
+
+int
+dtls_ecdsa_verify_sig_hash2(const unsigned char *pub_key, size_t key_size,
+                            const unsigned char *sign_hash, size_t sign_hash_size,
+                            dtls_ecdh_curve curve,
+                            unsigned char *result_r, unsigned char *result_s) {
   uint8_t sign[2 * DTLS_EC_KEY_SIZE];
+  uECC_Curve uecc_curve;
+  int curve_size;
+  if (!get_uecc_curve(curve, &uecc_curve)) {
+    dtls_warn("curve %" PRIu16 " not supported\n", curve);
+    return -1;
+  }
   (void)result_r;
   (void)result_s;
 
-  assert(key_size == DTLS_EC_KEY_SIZE);
-  assert(sign_hash_size >= uECC_BYTES);
-  assert(sizeof(sign) >= 2 * uECC_BYTES);
+  curve_size = uECC_curve_public_key_size(uecc_curve);
+
+  assert(key_size == (unsigned int)curve_size);
```

Comment:
That fires on my interop test. Adding 

```
dtls_warn("%d != %d\n", (int) key_size, curve_size);
```

I get 

```
Feb 26 16:46:45 WARN 32 != 64
tinydtls-server: crypto.c:638: dtls_ecdsa_verify_sig_hash2: Assertion `key_size == (unsigned int)curve_size' failed.
```

AFAIK, the `key-size` is passed in as `sizeof(config->keyx.ecdsa.other_pub_x)` in dtls.c, line 2589

Reply:
Beside the assert, the first interop-tests with Californium are successful.
Great!